tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_bluetooth.py

```diff
@@ -33,7 +32,10 @@ def test_toggle_bluetooth_settings(self):
 
         self.assertEquals(bluetooth_settings.device_name, device_name)
 
-        bluetooth_settings.enable_visible_to_all()
+        bluetooth_settings.enable_visible_to_all(self.data_layer)
 
         self.assertTrue(self.data_layer.bluetooth_is_discoverable)
         self.assertEquals(self.data_layer.bluetooth_name, device_name)
+
+        bluetooth_settings.disable_bluetooth()
+        self.assertFalse(self.data_layer.bluetooth_is_enabled)
```

Comment:
I guess if bluetooth is disabled at the start, we need to have a tearDown function in which we disable bluetooth.
The tearDown function is always called, even when the test is failing somewhere.